src/utils/tests/scopes.js

```diff
@@ -103,4 +106,251 @@ describe("scopes", () => {
+                writable: true
+              },
+              arguments: {
+                enumerable: true,
+                configurable: false,
+                value: {
+                  type: "object",
+                  class: "Arguments",
+                  actor: "server2.conn1.pausedobj46",
+                  extensible: true,
+                  frozen: false,
+                  sealed: false,
+                  ownPropertyLength: 3,
+                  preview: {
+                    kind: "Object",
+                    ownProperties: {},
+                    ownPropertiesLength: 3,
+                    safeGetterValues: {}
+                  }
+                },
+                writable: true
+              }
+            }
+          }
+        },
+        function: {
+          frozen: false,
+          name: "func",
+          displayName: "func",
+          actor: "server2.conn1.pausedobj45",
+          location: {
+            url: "file://script.js",
+            line: 8
+          },
+          class: "Function",
+          type: "object",
+          extensible: true,
+          sealed: false,
+          parameterNames: [ "b", "d" ]
+        },
+        bindings: {
+          arguments: [
+            {
+              b: {
+                enumerable: true,
+                configurable: false,
+                value: "bb",
+                writable: true
+              }
+            },
+            {
+              d: {
+                enumerable: true,
+                configurable: false,
+                value: "dd",
+                writable: true
+              }
+            }
+          ],
+          variables: {
+            c: {
+              enumerable: true,
+              configurable: false,
+              value: "cc",
+              writable: true
+            },
+            arguments: {
+              enumerable: true,
+              configurable: false,
+              value: {
+                type: "object",
+                class: "Arguments",
+                actor: "server2.conn1.pausedobj47",
+                extensible: true,
+                frozen: false,
+                sealed: false,
+                ownPropertyLength: 5,
+                preview: {
+                  kind: "Object",
+                  ownProperties: {},
+                  ownPropertiesLength: 5,
+                  safeGetterValues: {}
+                }
+              },
+              writable: true
+            }
+          }
+        }
+      }
+    };
+
+    let pauseInfo;
+
+    beforeEach(function() {
+      // Default pauseInfo is using the innermost frame in the stack.
+      pauseInfo = fromJS({
+        why: {
+          type: "debuggerStatement"
+        },
+        frame,
+        isInterrupted: false
+      });
+
+      global.L10N = { getStr: () => "" };
+    });
+
+    afterEach(function() {
+      delete global.L10N;
+    });
+
+    it("Returns variables from the outer scope", function() {
```

Comment:
Ideally I'd like a test to return variable from a frame higher in the call stack, but I've been lazy as this would need a completely different `pauseInfo`.

Which made me wonder why we need the frame after all because it seems everything we need is actually in `pauseInfo` ?